hooks/useCohort/useCohort.tsx

```diff
@@ -0,0 +1,58 @@
+import { createContext, useState, useEffect, useMemo, useContext } from "react";
+import { ApiServiceBuilder } from "@/helpers/api";
+import { HTTP_METHOD } from "@/types/api";
+import {
+  ICohort,
+  CohortProviderProps,
+  DEFAULT_COHORT_YEAR,
+} from "@/types/useCohort";
+
+const CohortContext = createContext<ICohort>({
+  currentCohortYear: DEFAULT_COHORT_YEAR,
+  loading: false,
+});
+
+export const CohortProvider = ({ children }: CohortProviderProps) => {
+  const [currentCohortYear, setCurrentCohortYear] =
+    useState<number>(DEFAULT_COHORT_YEAR);
+  const [loading, setLoading] = useState<boolean>(true);
```

Comment:
```suggestion
  const [loading, setLoading] = useState(false);
```
default should be false because the useEffect sets loading as true